src/test/java/com/lafaspot/pop/client/PopClientProdIT.java

```diff
@@ -47,6 +55,74 @@ public void init() throws PopException {
+        f.get();
+        final Channel ch = sess.getChannel();
+        Assert.assertNotNull(ch);
+        final ChannelPipeline pipe = ch.pipeline();
+        Assert.assertNotNull(pipe);
+        final String[] handlerList = { PopSession.SSL_HANDLER, PopSession.INACTIVITY_HANDLER, PopSession.DELIMITER, PopSession.DECODER,
+                PopSession.ENCODER, PopSession.POP_HANDLER };
+        final Iterator<Entry<String, ChannelHandler>> it = pipe.iterator();
+        int index = 0;
+        while (it.hasNext()) {
+            final Entry<String, ChannelHandler> entry = it.next();
+            Assert.assertNotNull(entry);
+            final String name = entry.getKey();
+            Assert.assertEquals(name, handlerList[index++]);
+            Assert.assertNotNull(entry.getValue());
+        }
+        System.out.println("testChannelPipeline DONE");
+    }
+
+    @Test
+    public void testChangeChannelPipeline() throws PopException, InterruptedException, ExecutionException {
+        System.out.println("testChangeChannelPipeline Begin");
+        PopSession sess = client.createSession();
+        PopFuture<PopCommandResponse> f = sess.connect(server, port, 120000, 120000);
+        f.get();
+        final Channel ch = sess.getChannel();
+        Assert.assertNotNull(ch);
+        final ChannelPipeline pipe = ch.pipeline();
+        Assert.assertNotNull(pipe);
+        final String[] handlerList = { PopSession.SSL_HANDLER, PopSession.INACTIVITY_HANDLER, PopSession.DELIMITER, PopSession.DECODER,
+                PopSession.ENCODER, PopSession.POP_HANDLER };
+        final Iterator<Entry<String, ChannelHandler>> it = pipe.iterator();
+        int index = 0;
+        while (it.hasNext()) {
+            final Entry<String, ChannelHandler> entry = it.next();
+            Assert.assertNotNull(entry);
+            final String name = entry.getKey();
+            Assert.assertEquals(name, handlerList[index++]);
+            Assert.assertNotNull(entry.getValue());
+        }
+
+        // Changing the pipeline by removing few handler and adding new handlers.
+        final ChannelHandler newHandler = new LineBasedFrameDecoder(1);
+        pipe.addAfter(PopSession.SSL_HANDLER, "NEW_HANDLER", newHandler);
+        pipe.remove(PopSession.INACTIVITY_HANDLER);
+        pipe.remove(PopSession.DELIMITER);
+        pipe.remove(PopSession.DECODER);
+        pipe.remove(PopSession.ENCODER);
+        pipe.remove(PopSession.POP_HANDLER);
+
+        final String[] newHandlerList = { PopSession.SSL_HANDLER, "NEW_HANDLER" };
+        final Iterator<Entry<String, ChannelHandler>> it2 = pipe.iterator();
+        int index2 = 0;
+        while (it2.hasNext()) {
+            final Entry<String, ChannelHandler> entry = it2.next();
+            Assert.assertNotNull(entry);
+            final String name = entry.getKey();
+            Assert.assertEquals(name, newHandlerList[index2++]);
+            Assert.assertNotNull(entry.getValue());
+        }
```

Comment:
the remove old + add new will never be atomic, test and  handle the case the following case:
- remove old
- channel disconnected by now
- you add new handlers

Reply:
Adding new hanlders doesn't affect. We should be careful inside the handler to handle inactive channels.

Reply:
where are the UTs I asked for?

Reply:
As previously stated, the the UTs for the use-case that you asked for, doesn't have any effect. So, it ll be the same as the previous UT.